From 75a02c586f6ec49d4316f79004834250ef42b0e0 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Sat, 25 Jul 2015 15:11:19 +0200
Subject: [PATCH] fix test

The previous version only worked with libxml2 < 2.9.2, because of a bug
in handling nested elements of the same name.

Change-Id: I1368cedf5845895154806d4ecf427bfbed19165f
---
 configure.ac            |    4 ++++
 src/test/importtest.cpp |    6 +++---
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1611dcb..a075e73 100644
--- a/configure.ac
+++ b/configure.ac
@@ -128,6 +128,10 @@ AS_IF([test "x$enable_tools" = "xyes"], [
 	PKG_CHECK_MODULES([REVENGE_STREAM],[
 		librevenge-stream-0.0
 	])
+        PKG_CHECK_EXISTS([libxml-2.0 >= 2.9.2],
+                [],
+                [AC_MSG_ERROR([libxml2 >= 2.9.2 is needed to run tests])]
+        )
 ])
 AC_SUBST([REVENGE_GENERATORS_CFLAGS])
 AC_SUBST([REVENGE_GENERATORS_LIBS])
diff --git a/src/test/importtest.cpp b/src/test/importtest.cpp
index 8554309..0296380 100644
--- a/src/test/importtest.cpp
+++ b/src/test/importtest.cpp
@@ -272,9 +272,9 @@ void ImportTest::testVsdxImportBgColorFromTheme()
 void ImportTest::testVsdxCharBgColor()
 {
   m_doc = parse("bgcolor.vsdx", m_buffer);
-  assertXPathNoAttribute(m_doc, "/document/page/layer/textObject/paragraph/span", "background-color");
-  assertXPath(m_doc, "/document/page/layer/layer[2]/textObject/paragraph/span", "background-color", "#9dbb61");
-  assertXPath(m_doc, "/document/page/layer/layer[2]/layer[2]/textObject/paragraph/span", "background-color", "#9dbb61");
+  assertXPathNoAttribute(m_doc, "/document/page/layer[1]/textObject/paragraph/span", "background-color");
+  assertXPath(m_doc, "/document/page/layer[2]/textObject/paragraph/span", "background-color", "#9dbb61");
+  assertXPath(m_doc, "/document/page/layer[3]/textObject/paragraph/span", "background-color", "#9dbb61");
 }
 
 CPPUNIT_TEST_SUITE_REGISTRATION(ImportTest);
-- 
1.7.9.5

